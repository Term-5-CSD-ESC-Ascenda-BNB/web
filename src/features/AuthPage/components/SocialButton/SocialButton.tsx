import { Button } from '@mantine/core';

interface SocialButtonProps {
  children: React.ReactNode;
  onClick?: () => void;
  disabled?: boolean;
}

export function SocialButton({ children, onClick, disabled }: SocialButtonProps) {
  return (
    <Button
      variant="default"
      radius="xl"
      px={0}
      size="lg"
      style={{ width: 54, height: 54, background: '#fff', border: '1px solid #e4e4ec' }}
      onClick={onClick}
      disabled={disabled}
    >
      {children}
    </Button>
  );
}
