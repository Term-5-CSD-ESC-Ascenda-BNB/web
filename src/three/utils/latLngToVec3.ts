import * as THREE from 'three';

export function latLngToVec3(lat: number, lng: number, radius: number): THREE.Vector3 {
  const elevation = THREE.MathUtils.degToRad(90 - lat);
  const azimuth = THREE.MathUtils.degToRad(lng) + Math.PI / 2; // Add PI/2 to align with the globe's coordinate system

  const position = new THREE.Vector3().setFromSphericalCoords(radius, elevation, azimuth);

  return position;
}
