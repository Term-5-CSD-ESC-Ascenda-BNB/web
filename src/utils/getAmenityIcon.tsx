import {
  IconWind,
  IconBriefcase,
  IconIroning3,
  IconPlugConnected,
  IconShirt,
  IconScissors,
  IconBath,
  IconUsersGroup,
  IconSwimming,
  IconParking,
  IconBell,
  IconScale,
  IconDeviceTv,
  IconPhone,
  IconWriting,
  IconCoffee,
  IconBed,
  IconBabyCarriage,
  IconTrowel,
  IconShoe,
  IconHelpCircle,
} from '@tabler/icons-react';

export function getAmenityIcon(label: string): React.ReactElement {
  const f = label.toLowerCase();

  if (f.includes('air')) return <IconWind size={16} />;
  if (f.includes('business')) return <IconBriefcase size={16} />;
  if (f.includes('iron')) return <IconIroning3 size={16} />;
  if (f.includes('data') || f.includes('port')) return <IconPlugConnected size={16} />;
  if (f.includes('dry')) return <IconShirt size={16} />;
  if (f.includes('hair')) return <IconScissors size={16} />;
  if (f.includes('meeting')) return <IconUsersGroup size={16} />;
  if (f.includes('pool')) return <IconSwimming size={16} />;
  if (f.includes('garage') || f.includes('parking')) return <IconParking size={16} />;
  if (f.includes('room service')) return <IconBell size={16} />;
  if (f.includes('safe')) return <IconScale size={16} />;
  if (f.includes('tv')) return <IconDeviceTv size={16} />;
  if (f.includes('phone')) return <IconPhone size={16} />;
  if (f.includes('desk') || f.includes('workspace')) return <IconWriting size={16} />;
  if (f.includes('coffee') || f.includes('tea') || f.includes('minibar') || f.includes('kettle'))
    return <IconCoffee size={16} />;
  if (f.includes('bed') || f.includes('bedding') || f.includes('linen'))
    return <IconBed size={16} />;
  if (f.includes('towel')) return <IconTrowel size={16} />;
  if (f.includes('crib') || f.includes('infant')) return <IconBabyCarriage size={16} />;
  if (f.includes('bath') || f.includes('soak')) return <IconBath size={16} />;
  if (f.includes('shower')) return <IconShoe size={16} />;

  return <IconHelpCircle size={16} />;
}
