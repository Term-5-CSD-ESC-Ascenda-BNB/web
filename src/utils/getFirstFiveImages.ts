import type { ImageDetails } from '@/types/Hotel';

/**
 * Extracts the first five image URLs from a hotel object.
 * @param imageDetails - The image details object containing prefix, suffix, and count.
 * @return An array of up to five image URLs.
 */
export function getFirstFiveImages(imageDetails: ImageDetails): string[] {
  if (!imageDetails || !imageDetails.prefix || !imageDetails.suffix) {
    return [];
  }

  const images: string[] = [];
  const count = Math.min(imageDetails.count, 5);

  for (let i = 0; i < count; i++) {
    images.push(`${imageDetails.prefix}${i}${imageDetails.suffix}`);
  }

  return images;
}
