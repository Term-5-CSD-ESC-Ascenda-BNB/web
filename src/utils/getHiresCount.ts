export function getHiresCount(hires_string: string): number {
  // separate the string by commas
  const hiresArray = hires_string
    .split(',')
    // filter out any empty strings
    .filter((value) => value.trim() !== '');
  // return the length of the array
  return hiresArray.length;
}
