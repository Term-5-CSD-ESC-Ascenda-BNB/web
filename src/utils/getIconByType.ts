import React from 'react';
import {
  IconBuildingSkyscraper,
  IconPlane,
  IconBuilding,
  IconBuildingBank,
  IconSchool,
  IconToolsKitchen2,
  IconShoppingBag,
  IconBuildingMonument,
  IconMask,
  IconBuildingStadium,
  IconCards,
  IconGolf,
  IconGlassFullFilled,
  IconBriefcase,
  IconBuildingHospital,
  IconBuildingCommunity,
  IconSkiJumping,
  IconAnchor,
  IconDirectionsFilled,
  IconTree,
  IconIceCream,
  IconSunglasses,
  IconBuildingEstate,
  IconMapPin,
  type IconProps,
} from '@tabler/icons-react';

/**
 * Returns the appropriate Tabler icon component for a destination type
 * @param type - The destination type (e.g., 'city', 'airport', 'museum')
 * @returns React component for the icon
 */
export const getIconByType = (type: string): React.FC<IconProps> => {
  const iconMap: Record<string, React.FC<IconProps>> = {
    city: IconBuildingSkyscraper,
    airport: IconPlane,
    hotel: IconBuilding,
    museum: IconBuildingBank,
    museums: IconBuildingBank,
    school: IconSchool,
    restaurant: IconToolsKitchen2,
    shopping: IconShoppingBag,
    historic: IconBuildingMonument,
    monument: IconBuildingMonument,
    theater: IconMask,
    stadium: IconBuildingStadium,
    casino: IconCards,
    golf: IconGolf,
    winery: IconGlassFullFilled,
    business: IconBriefcase,
    medical: IconBuildingHospital,
    civic: IconBuildingCommunity,
    skiing: IconSkiJumping,
    anchor: IconAnchor,
    sign: IconDirectionsFilled,
    tree: IconTree,
    icecream: IconIceCream,
    sunglass: IconSunglasses,
    district: IconBuildingEstate,
  };
  return iconMap[type.toLowerCase()] || IconMapPin;
};
