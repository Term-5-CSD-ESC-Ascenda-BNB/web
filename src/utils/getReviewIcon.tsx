import {
  IconMapPin,
  IconBed,
  IconWifi,
  IconCoffee,
  IconToolsKitchen2,
  IconArmchair,
  IconHeartHandshake,
  IconMoodSmile,
  IconHelpCircle,
  IconUser,
  IconUsersGroup,
  IconBriefcase,
  IconHeart,
} from '@tabler/icons-react';
import React from 'react';

export function getReviewIcon(label: string | undefined): React.ReactElement {
  if (!label) return <IconHelpCircle size={16} />;

  const l = label.toLowerCase();

  // Category scores
  if (l.includes('location')) return <IconMapPin size={16} />;
  if (l.includes('room')) return <IconBed size={16} />;
  if (l.includes('wifi') || l.includes('internet')) return <IconWifi size={16} />;
  if (l.includes('breakfast')) return <IconCoffee size={16} />;
  if (l.includes('food') || l.includes('dining')) return <IconToolsKitchen2 size={16} />;
  if (l.includes('comfort')) return <IconArmchair size={16} />;
  if (l.includes('service')) return <IconHeartHandshake size={16} />;
  if (l.includes('vibe')) return <IconMoodSmile size={16} />;

  // Traveler types
  if (l.includes('solo')) return <IconUser size={16} />;
  if (l.includes('couple')) return <IconHeart size={16} />;
  if (l.includes('family')) return <IconUsersGroup size={16} />;
  if (l.includes('business')) return <IconBriefcase size={16} />;

  return <IconHelpCircle size={16} />;
}
