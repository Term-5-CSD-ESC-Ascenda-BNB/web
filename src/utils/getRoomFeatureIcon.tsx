import {
  IconRulerMeasure,
  IconUsers,
  IconBed,
  IconWifi,
  IconEye,
  IconDeviceTv,
  IconBath,
  IconHelpCircle,
} from '@tabler/icons-react';
import React from 'react';

export function getRoomFeatureIcon(label: string | undefined): React.ReactElement {
  if (!label) return <IconHelpCircle size={14} />;

  const f = label.toLowerCase();

  if (f.includes('sq ft') || f.includes('sqm')) return <IconRulerMeasure size={14} />;
  if (f.includes('sleep') || f.includes('occupancy')) return <IconUsers size={14} />;
  if (f.includes('bed') || f.includes('king') || f.includes('twin') || f.includes('double'))
    return <IconBed size={14} />;
  if (f.includes('wifi') || f.includes('internet')) return <IconWifi size={14} />;
  if (f.includes('view') || f.includes('city') || f.includes('courtyard'))
    return <IconEye size={14} />;
  if (f.includes('tv') || f.includes('lcd') || f.includes('inch'))
    return <IconDeviceTv size={14} />;
  if (f.includes('bath') || f.includes('bathtub') || f.includes('shower'))
    return <IconBath size={14} />;

  return <IconHelpCircle size={14} />;
}
