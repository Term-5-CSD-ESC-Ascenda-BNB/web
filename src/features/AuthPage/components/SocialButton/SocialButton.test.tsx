import { render, screen, userEvent } from '@/tests/utils';
import { describe, it, expect, vi } from 'vitest';
import { SocialButton } from './SocialButton';

describe('SocialButton', () => {
  describe('rendering', () => {
    it('renders without crashing and matches snapshot', () => {
      const { container } = render(
        <SocialButton>
          <span>Icon</span>
        </SocialButton>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders children content', () => {
      render(
        <SocialButton>
          <span data-testid="social-icon">Google Icon</span>
        </SocialButton>
      );

      expect(screen.getByTestId('social-icon')).toBeInTheDocument();
      expect(screen.getByText('Google Icon')).toBeInTheDocument();
    });

    it('renders with complex children', () => {
      render(
        <SocialButton>
          <div>
            <svg data-testid="svg-icon">
              <circle />
            </svg>
            <span>Complex Content</span>
          </div>
        </SocialButton>
      );

      expect(screen.getByTestId('svg-icon')).toBeInTheDocument();
      expect(screen.getByText('Complex Content')).toBeInTheDocument();
    });
  });

  describe('styling and appearance', () => {
    it('applies correct button styles', () => {
      render(
        <SocialButton>
          <span>Icon</span>
        </SocialButton>
      );

      const button = screen.getByRole('button');

      // Check inline styles
      expect(button).toHaveStyle({
        width: '54px',
        height: '54px',
        background: '#fff',
        border: '1px solid #e4e4ec',
      });
    });

    it('has correct Mantine props applied', () => {
      const { container } = render(
        <SocialButton>
          <span>Icon</span>
        </SocialButton>
      );

      const button = container.querySelector('button');

      // Mantine applies these as CSS classes, so we check for their presence
      expect(button).toHaveClass('mantine-Button-root');
    });
  });

  describe('interactions', () => {
    it('calls onClick handler when clicked', async () => {
      const handleClick = vi.fn();
      const user = userEvent.setup();

      render(
        <SocialButton onClick={handleClick}>
          <span>Icon</span>
        </SocialButton>
      );

      const button = screen.getByRole('button');
      await user.click(button);

      expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it('calls onClick handler multiple times on multiple clicks', async () => {
      const handleClick = vi.fn();
      const user = userEvent.setup();

      render(
        <SocialButton onClick={handleClick}>
          <span>Icon</span>
        </SocialButton>
      );

      const button = screen.getByRole('button');
      await user.click(button);
      await user.click(button);
      await user.click(button);

      expect(handleClick).toHaveBeenCalledTimes(3);
    });

    it('does not crash when onClick is not provided', async () => {
      const user = userEvent.setup();

      render(
        <SocialButton>
          <span>Icon</span>
        </SocialButton>
      );

      const button = screen.getByRole('button');

      // Should not throw an error
      await expect(user.click(button)).resolves.not.toThrow();
    });

    it('supports keyboard interactions', async () => {
      const handleClick = vi.fn();
      const user = userEvent.setup();

      render(
        <SocialButton onClick={handleClick}>
          <span>Icon</span>
        </SocialButton>
      );

      const button = screen.getByRole('button');
      button.focus();
      await user.keyboard('{Enter}');

      expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it('supports spacebar activation', async () => {
      const handleClick = vi.fn();
      const user = userEvent.setup();

      render(
        <SocialButton onClick={handleClick}>
          <span>Icon</span>
        </SocialButton>
      );

      const button = screen.getByRole('button');
      button.focus();
      await user.keyboard(' ');

      expect(handleClick).toHaveBeenCalledTimes(1);
    });
  });

  describe('disabled state', () => {
    it('renders as disabled when disabled prop is true', () => {
      render(
        <SocialButton disabled>
          <span>Icon</span>
        </SocialButton>
      );

      const button = screen.getByRole('button');
      expect(button).toBeDisabled();
    });

    it('does not call onClick when disabled and clicked', async () => {
      const handleClick = vi.fn();
      const user = userEvent.setup();

      render(
        <SocialButton onClick={handleClick} disabled>
          <span>Icon</span>
        </SocialButton>
      );

      const button = screen.getByRole('button');
      await user.click(button);

      expect(handleClick).not.toHaveBeenCalled();
    });

    it('is not disabled by default', () => {
      render(
        <SocialButton>
          <span>Icon</span>
        </SocialButton>
      );

      const button = screen.getByRole('button');
      expect(button).not.toBeDisabled();
    });

    it('can be enabled after being disabled', () => {
      const { rerender } = render(
        <SocialButton disabled>
          <span>Icon</span>
        </SocialButton>
      );

      let button = screen.getByRole('button');
      expect(button).toBeDisabled();

      rerender(
        <SocialButton disabled={false}>
          <span>Icon</span>
        </SocialButton>
      );

      button = screen.getByRole('button');
      expect(button).not.toBeDisabled();
    });
  });

  describe('accessibility', () => {
    it('is focusable', () => {
      render(
        <SocialButton>
          <span>Icon</span>
        </SocialButton>
      );

      const button = screen.getByRole('button');
      button.focus();
      expect(button).toHaveFocus();
    });

    it('has correct role', () => {
      render(
        <SocialButton>
          <span>Icon</span>
        </SocialButton>
      );

      expect(screen.getByRole('button')).toBeInTheDocument();
    });

    it('is accessible by screen readers', () => {
      render(
        <SocialButton>
          <span>Google Sign In</span>
        </SocialButton>
      );

      // The button should be accessible with its text content
      expect(screen.getByRole('button', { name: 'Google Sign In' })).toBeInTheDocument();
    });

    it('supports tabbing navigation', () => {
      render(
        <div>
          <SocialButton>
            <span>Button 1</span>
          </SocialButton>
          <SocialButton>
            <span>Button 2</span>
          </SocialButton>
        </div>
      );

      const button1 = screen.getByRole('button', { name: 'Button 1' });
      const button2 = screen.getByRole('button', { name: 'Button 2' });

      button1.focus();
      expect(button1).toHaveFocus();

      // Simulate tab navigation
      button2.focus();
      expect(button2).toHaveFocus();
    });
  });

  describe('edge cases', () => {
    it('handles empty children', () => {
      render(<SocialButton>{null}</SocialButton>);

      const button = screen.getByRole('button');
      expect(button).toBeInTheDocument();

      // The button contains Mantine's internal structure but no user content
      const buttonLabel = button.querySelector('.mantine-Button-label');
      expect(buttonLabel).toBeEmptyDOMElement();
    });

    it('handles undefined children', () => {
      render(<SocialButton>{undefined}</SocialButton>);

      const button = screen.getByRole('button');
      expect(button).toBeInTheDocument();
    });

    it('handles boolean children', () => {
      render(<SocialButton>{false}</SocialButton>);

      const button = screen.getByRole('button');
      expect(button).toBeInTheDocument();
    });

    it('handles number children', () => {
      render(<SocialButton>{42}</SocialButton>);

      const button = screen.getByRole('button');
      expect(button).toBeInTheDocument();
      expect(screen.getByText('42')).toBeInTheDocument();
    });
  });

  describe('component integration', () => {
    it('works with real social media icons', () => {
      const GoogleIcon = () => (
        <svg data-testid="google-icon" width="20" height="20" viewBox="0 0 24 24">
          <path d="M22.56 12.25c0-.78-.07-1.53-.2-2.25H12v4.26h5.92c-.26 1.37-1.04 2.53-2.21 3.31v2.77h3.57c2.08-1.92 3.28-4.74 3.28-8.09z" />
        </svg>
      );

      render(
        <SocialButton>
          <GoogleIcon />
        </SocialButton>
      );

      expect(screen.getByTestId('google-icon')).toBeInTheDocument();
    });

    it('integrates well with click handlers for authentication', async () => {
      const mockAuth = vi.fn().mockResolvedValue({ success: true });
      const user = userEvent.setup();

      const handleGoogleAuth = () => {
        void mockAuth('google');
      };

      render(
        <SocialButton onClick={handleGoogleAuth}>
          <span>G</span>
        </SocialButton>
      );

      const button = screen.getByRole('button');
      await user.click(button);

      expect(mockAuth).toHaveBeenCalledWith('google');
    });
  });
});
