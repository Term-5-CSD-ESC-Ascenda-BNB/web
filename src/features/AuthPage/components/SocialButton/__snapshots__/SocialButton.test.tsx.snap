// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SocialButton > rendering > renders without crashing and matches snapshot 1`] = `
<div>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: DM Sans, sans-serif;--mantine-font-family-headings: Libre Franklin, serif;--mantine-primary-color-filled: var(--mantine-color-primary-filled);--mantine-primary-color-filled-hover: var(--mantine-color-primary-filled-hover);--mantine-primary-color-light: var(--mantine-color-primary-light);--mantine-primary-color-light-hover: var(--mantine-color-primary-light-hover);--mantine-primary-color-light-color: var(--mantine-color-primary-light-color);--mantine-primary-color-0: var(--mantine-color-primary-0);--mantine-primary-color-1: var(--mantine-color-primary-1);--mantine-primary-color-2: var(--mantine-color-primary-2);--mantine-primary-color-3: var(--mantine-color-primary-3);--mantine-primary-color-4: var(--mantine-color-primary-4);--mantine-primary-color-5: var(--mantine-color-primary-5);--mantine-primary-color-6: var(--mantine-color-primary-6);--mantine-primary-color-7: var(--mantine-color-primary-7);--mantine-primary-color-8: var(--mantine-color-primary-8);--mantine-primary-color-9: var(--mantine-color-primary-9);--mantine-color-primary-0: #F5F6F9;--mantine-color-primary-1: #E8EAF1;--mantine-color-primary-2: #BAC2D6;--mantine-color-primary-3: #99A4C1;--mantine-color-primary-4: #808AB1;--mantine-color-primary-5: #6E74A2;--mantine-color-primary-6: #626693;--mantine-color-primary-7: #545579;--mantine-color-primary-8: #43455E;--mantine-color-primary-9: #2E2E3D;--mantine-color-secondary-0: #F8F6F4;--mantine-color-secondary-1: #DCD6CC;--mantine-color-secondary-2: #C8BEB0;--mantine-color-secondary-3: #AD9D8A;--mantine-color-secondary-4: #9C8773;--mantine-color-secondary-5: #8F7867;--mantine-color-secondary-6: #786256;--mantine-color-secondary-7: #635249;--mantine-color-secondary-8: #51433D;--mantine-color-secondary-9: #2B231F;}:root[data-mantine-color-scheme="dark"]{--mantine-color-anchor: var(--mantine-color-primary-4);--mantine-color-primary-text: var(--mantine-color-primary-4);--mantine-color-primary-filled: var(--mantine-color-primary-8);--mantine-color-primary-filled-hover: var(--mantine-color-primary-9);--mantine-color-primary-light: rgba(98, 102, 147, 0.15);--mantine-color-primary-light-hover: rgba(98, 102, 147, 0.2);--mantine-color-primary-light-color: var(--mantine-color-primary-3);--mantine-color-primary-outline: var(--mantine-color-primary-4);--mantine-color-primary-outline-hover: rgba(128, 138, 177, 0.05);--mantine-color-secondary-text: var(--mantine-color-secondary-4);--mantine-color-secondary-filled: var(--mantine-color-secondary-8);--mantine-color-secondary-filled-hover: var(--mantine-color-secondary-9);--mantine-color-secondary-light: rgba(120, 98, 86, 0.15);--mantine-color-secondary-light-hover: rgba(120, 98, 86, 0.2);--mantine-color-secondary-light-color: var(--mantine-color-secondary-3);--mantine-color-secondary-outline: var(--mantine-color-secondary-4);--mantine-color-secondary-outline-hover: rgba(156, 135, 115, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-anchor: var(--mantine-color-primary-6);--mantine-color-primary-text: var(--mantine-color-primary-filled);--mantine-color-primary-filled: var(--mantine-color-primary-6);--mantine-color-primary-filled-hover: var(--mantine-color-primary-7);--mantine-color-primary-light: rgba(98, 102, 147, 0.1);--mantine-color-primary-light-hover: rgba(98, 102, 147, 0.12);--mantine-color-primary-light-color: var(--mantine-color-primary-6);--mantine-color-primary-outline: var(--mantine-color-primary-6);--mantine-color-primary-outline-hover: rgba(98, 102, 147, 0.05);--mantine-color-secondary-text: var(--mantine-color-secondary-filled);--mantine-color-secondary-filled: var(--mantine-color-secondary-6);--mantine-color-secondary-filled-hover: var(--mantine-color-secondary-7);--mantine-color-secondary-light: rgba(120, 98, 86, 0.1);--mantine-color-secondary-light-hover: rgba(120, 98, 86, 0.12);--mantine-color-secondary-light-color: var(--mantine-color-secondary-6);--mantine-color-secondary-outline: var(--mantine-color-secondary-6);--mantine-color-secondary-outline-hover: rgba(120, 98, 86, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <button
    class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
    data-size="lg"
    data-variant="default"
    style="--button-height: var(--button-height-lg); --button-padding-x: var(--button-padding-x-lg); --button-fz: var(--mantine-font-size-lg); --button-radius: var(--mantine-radius-xl); --button-bg: var(--mantine-color-default); --button-hover: var(--mantine-color-default-hover); --button-color: var(--mantine-color-default-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-default-border); width: 54px; height: 54px; background: rgb(255, 255, 255); border: 1px solid rgb(228, 228, 236); padding-inline: 0rem;"
    type="button"
  >
    <span
      class="m_80f1301b mantine-Button-inner"
    >
      <span
        class="m_811560b9 mantine-Button-label"
      >
        <span>
          Icon
        </span>
      </span>
    </span>
  </button>
</div>
`;
